integration-libs/opf/root/components/opf-payment-verification/opf-payment-verification.service.ts

```diff
@@ -33,4 +56,82 @@ export class OpfPaymentVerificationService {
   goToPage(cxRoute: string) {
     return this.routingService.go({ cxRoute });
   }
+
+  verifyResultUrl(route: ActivatedRoute): Observable<{
+    paymentSessionId: string;
+    responseMap: OpfResponseMapElement[];
+  }> {
+    return route?.routeConfig?.data?.cxRoute === 'paymentVerificationResult'
+      ? route.queryParams.pipe(
+          concatMap((params: Params) => {
+            if (!params) {
+              return throwError(this.defaultError);
+            }
+
+            const responseMap: OpfResponseMapElement[] =
+              this.getOpfResponseMap(params);
+
+            const paymentSessionId = this.findInOpfResponseMap(
+              OpfPaymenVerificationUrlInput.PAYMENT_SESSION_ID,
+              responseMap
+            );
+            if (!paymentSessionId) {
+              return throwError(this.defaultError);
+            }
+            return of({
+              paymentSessionId,
+              responseMap,
+            });
+          })
+        )
+      : throwError({
+          ...this.defaultError,
+          message: 'opf.checkout.errors.cancelPayment',
+        });
+  }
+
+  placeOrder() {
+    return this.opfOrderFacade.placeOpfOrder(true);
+  }
+
+  verifyPayment(
+    paymentSessionId: string,
+    responseMap: OpfResponseMapElement[]
+  ): Observable<boolean> {
+    return this.opfCheckoutService
+      .verifyPayment(paymentSessionId, {
+        responseMap: [...responseMap],
+      })
+      .pipe(
+        concatMap((response: OpfPaymentVerificationResponse) =>
+          this.isPaymentSuccessful(response)
+        )
+      );
+  }
+
+  isPaymentSuccessful(response: OpfPaymentVerificationResponse) {
+    if (
+      response.result === OpfPaymentVerificationResult.AUTHORIZED ||
+      response.result === OpfPaymentVerificationResult.DELAYED
+    ) {
+      return of(true);
```

Comment:
I set this function return Observable<boolean> but technically, it never returns false. It is either of(true) or throwError.
Value 'true' is never read, we could also return EMPTY or of(}), it does not matter in my opinion.